auth-service/src/main/java/com/auth/authservice/datalayer/roles/Role.java

```diff
@@ -21,18 +21,10 @@
 @Builder(toBuilder = true)
 public class Role {
 
-    public Role(long id, String name) {
-        this.id = id;
-        this.name = name;
-        this.parent = null;
-    }
 
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private long id;
     private String name;
 
-    @ManyToOne
-    @JoinColumn(name = "parent_id")
-    private Role parent;
```

Comment:
Great now that the structure of roles is completely flat we have more freedom in our usage of the role.